javawrapper/device/iothub-client-javawrapper/Message.java

```diff
@@ -0,0 +1,123 @@
+// Copyright(c) Microsoft.All rights reserved. 
+// Licensed under the MIT license.See LICENSE file in the project root for full license information. 
+
+
+package javaWrapper;
+
+import com.sun.jna.ptr.IntByReference;
+import com.sun.jna.ptr.PointerByReference;
+
+import javaWrapper.Iothub_client_wrapperLibrary.IOTHUB_MESSAGE_HANDLE;
+import javaWrapper.Iothub_client_wrapperLibrary.IOTHUB_MESSAGE_RESULT;
+import javaWrapper.Iothub_client_wrapperLibrary.MAP_HANDLE;
+
+public class Message 
+{
+	public static IOTHUB_MESSAGE_HANDLE messageHandle = null;
+	private static MAP_HANDLE propMap;
+	
+	
+	public Message()
+	{
+		
+	}
+	
+	public Message(byte[] body) 
+    {
+        if (body == null) {
+            throw new IllegalArgumentException("Message body cannot be 'null'.");
+        }      
+
+        messageHandle = Iothub_client_wrapperLibrary.INSTANCE.IoTHubMessage_CreateFromByteArray(body, body.length);
+    }
+    
+	public Message(String body) 
+	{
+		if (body == null) 
+		{
+			throw new IllegalArgumentException("Message body cannot be 'null'.");
+		}      
+
+		messageHandle = Iothub_client_wrapperLibrary.INSTANCE.IoTHubMessage_CreateFromString(body);
+	}
+    
+	public IOTHUB_MESSAGE_HANDLE clone()
+	{
+		return Iothub_client_wrapperLibrary.INSTANCE.IoTHubMessage_Clone(messageHandle);
+	}
+    
+	public byte[] getBytes()
+	{
+		String body;
+        
+		body = Iothub_client_wrapperLibrary.INSTANCE.IoTHubMessage_GetString(messageHandle);
+		
+		return body.getBytes();
+	}
+    
+	public MAP_HANDLE getProperties() 
+	{
+		return Iothub_client_wrapperLibrary.INSTANCE.IoTHubMessage_Properties(messageHandle);	
+	}
+    
+	public void setProperty(String name, String value) 
+	{    
+		if (name == null) 
+		{
+			throw new IllegalArgumentException("Property name cannot be 'null'.");
+		}
+
+		if (value == null) 
+		{
+			throw new IllegalArgumentException("Property value cannot be 'null'.");
+		}
+
+		propMap = this.getProperties();
+		Map map = new Map(propMap);
```

Comment:
Every time you're adding a new property you're creating a new Map. Shouldn't this be the same Map?